src/CashuWallet.ts

```diff
@@ -440,6 +442,19 @@ class CashuWallet {
 			keys.id,
 			options?.counter
 		);
+		if (options?.privkey != undefined) {
+			proofsToSend = getSignedProofs(
+				proofsToSend.map((p) => {
+					return {
+						amount: p.amount,
+						C: pointFromHex(p.C),
+						id: p.id,
+						secret: new TextEncoder().encode(p.secret)
+					};
```

Comment:
Could we refactor this into `getSignedProofs`?

Reply:
Just saw it's in cashu/crypto and the `Proof` type has a different definition:

`@cashu/cashu-ts`

```ts
 */
export type Proof = {
	id: string;
	amount: number;
	secret: string;
	C: string;
};
```

vs `@cashu/crypto`:

```ts
export type Proof = {
	C: ProjPointType<bigint>;
	secret: Uint8Array;
	amount: number;
	id: string;
	witness?: Witness;
};
```

☠️

Reply:
Yes, thats not great. Unrelated to this PR, but we should add a utility that hides this back-and-forth between the two formats in the near future